### ISSUE TITLE
`Raster.data` is not consistently a masked_array

### ISSUE THREAD

**Issue by erikmannerfelt** [GlacioHack/geoutils#231]
**Describe the bug**
There is no check in the data setter that the data is a masked array, so arguably weird situations can happen as a consequence.

**To Reproduce**
```python
>>> import geoutils as gu
>>> import numpy as np
>>> import rasterio as rio
>>> data = np.ones((5, 5), dtype="int32")
>>> transform = rio.transform.from_origin(0, 5, 1, 1)
>>> img = gu.Raster.from_array(data, transform=transform, crs=4326)
>>> # Now, the data has been converted to a masked_array
>>> img.data.mask
False
>>> img.data = data.reshape(img.data.shape)
>>> # If the data is assigned to again, it's not necessarily a masked_array
>>> img.data.mask
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'numpy.ndarray' object has no attribute 'mask'
```

**Expected behavior**
The data setter should make sure that the data type is consistent. Right now, any class that inherits from `np.ndarray` is allowed!

**Comment by adehecq** [comment]
Good point! I will try to look at it before my holidays.

**Comment by adehecq** [comment]
Seems to be fixed as of PR #289 or #265.
